### PR TITLE
MSI-2549-Disable-possibility-to-place-In-Store-Pickup-orders-for-default-source.

### PR DESCRIPTION
**Pull request by swnsma**
### Description (*)
Prevent save Default Source as Pickup Location.

### Fixed Issues (if relevant)
1. https://github.com/magento/inventory/issues/2549: Disable possibility to place In-Store Pickup orders for default source.
### Manual testing scenarios (*)
1. Go to Default Source.
2. Try to save it as Pickup Location.
3. You can even try to do so with webapi.

### Questions or comments
Possible, this is paranoia, but should we validate Quote before submit if it uses Default Source for In-Store Pickup?
This is still a possible case to mark it as Pickup Location via DB.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/InventoryInStorePickup/etc/di.xml [hunk 1]
```diff
@@ -53,5 +53,12 @@
     <preference for="Magento\InventorySourceSelectionApi\Model\GetSourceItemQtyAvailableInterface" type="Magento\InventoryInStorePickup\Model\SourceSelection\GetSourceItemQtyAvailableService"/>
     <preference for="Magento\InventoryInStorePickupApi\Api\IsStorePickupOrderInterface" type="Magento\InventoryInStorePickup\Model\IsStorePickupOrder" />
     <preference for="Magento\InventoryInStorePickupApi\Api\GetPickupLocationInterface" type="Magento\InventoryInStorePickup\Model\GetPickupLocation" />
-    <preference for="Magento\InventoryInStorePickupApi\Api\Data\SearchResultInterface" type="Magento\InventoryInStorePickup\Model\SearchResult"/>
+    <preference for="Magento\InventoryInStorePickupApi\Api\Data\SearchResultInterface" type="Magento\InventoryInStorePickup\Model\SearchResult"/><type name="Magento\InventoryApi\Model\SourceValidatorChain">
```
Comment (novikor) [inline app/code/Magento/InventoryInStorePickup/etc/di.xml]: Move `<type` to the next line!

**Review by novikor — CHANGES_REQUESTED** [review]

File: app/code/Magento/InventoryInStorePickup/etc/di.xml (hunk 1 above)
Comment (swnsma) [reply app/code/Magento/InventoryInStorePickup/etc/di.xml]: Done.

File: app/code/Magento/InventoryInStorePickup/Model/Source/Validator/DefaultSourceUseForPickupLocationValidator.php [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function validate(SourceInterface $source): ValidationResult
+    {
+        $errors = [];
+
+        if ($this->isDefaultSource($source) && $this->isUsedAsPickupLocation($source)) {
```
Comment (ishakhsuvarov) [inline app/code/Magento/InventoryInStorePickup/Model/Source/Validator/DefaultSourceUseForPickupLocationValidator.php]: Is it really needed to have 2 one-line private methods?

Comment (swnsma) [reply app/code/Magento/InventoryInStorePickup/Model/Source/Validator/DefaultSourceUseForPickupLocationValidator.php]: It has look more clear.
Removed.

**Review by ishakhsuvarov — APPROVED** [review]

**Pull request merged**